library/alloc/src/raw_vec/mod.rs

```diff
@@ -465,7 +466,9 @@ impl<A: Allocator> RawVecInner<A> {
 
         // Don't allocate here because `Drop` will not deallocate when `capacity` is 0.
         if layout.size() == 0 {
-            return Ok(Self::new_in(alloc, elem_layout.align()));
+            // SAFETY: `Layout.align` is a non-zero power of two.
+            let align = unsafe { Alignment::new_unchecked(elem_layout.align()) };
```

Comment:
`Alignment::new_unchecked` has the same problem – it inserts an UB check. I think your best bet would be to add a perma-unstable method on `Layout` that allows accessing the inner `align` field.

Reply:
I used `std_internal` for the perma-unstable feature gate, is that appropriate?

Reply:
Yeah, that's fine!